app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -295,13 +295,42 @@ private void startPokeFly() {
         Intent intent = Pokefly.createIntent(this, trainerLevel, statusBarHeight, batterySaver);
         startService(intent);
 
-        pokeFlyRunning = true;
-
         if (settings.shouldLaunchPokemonGo()) {
             openPokemonGoApp();
         }
     }
 
+    private boolean isPokeFlyRunning() {
+        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo serviceInfo : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (Pokefly.class.getName().equals(serviceInfo.service.getClassName())) {
+                return true;
+            }
+        }
```

Comment:
@Blaisorblade  I'm saying a boolean and static function inside pokefly. I don't think that was the case before. Also, this looping code will still have the race conditions that a boolean might. A flag inside Pokefly guarantees that it'll flow the service state pretty accurately.